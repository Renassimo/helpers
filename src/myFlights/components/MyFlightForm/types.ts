import { MyFlightAttributes } from '@/myFlights/types';
import { Matcher } from '@/common/types/matchers';

export default interface MyFlightFormFieldProps {
  name: keyof MyFlightAttributes;
  label: string;
  options?: string[];
  matchers?: Matcher;
  type?: string;
  disabled?: boolean;
  isDate?: boolean;
}
