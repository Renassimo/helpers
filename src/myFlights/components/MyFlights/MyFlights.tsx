import Typography from '@mui/material/Typography';

import FlightsTable from '@/myFlights/components/FlightsTable';

const MyFlights = () => {
  return (
    <>
      <Typography component="h1" variant="h6" mt={1} textAlign="center">
        My Flights
      </Typography>
      <FlightsTable />
    </>
  );
};

export default MyFlights;
