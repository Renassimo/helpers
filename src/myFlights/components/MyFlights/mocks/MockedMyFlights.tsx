const MockedMyFlights = jest.fn(() => <div>MockedMyFlights</div>);

export default MockedMyFlights;
