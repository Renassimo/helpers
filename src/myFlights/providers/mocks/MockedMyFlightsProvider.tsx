const MyFlightsProvider = jest.fn(({ children }) => (
  <div>MyFlightsProvider - {children}</div>
));

export default MyFlightsProvider;
