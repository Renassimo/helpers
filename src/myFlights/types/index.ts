export * from './myFlightTypes';
export * from './myFlightsDb';
export * from './props';
