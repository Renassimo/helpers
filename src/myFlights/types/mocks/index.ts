export * from './flights';
export * from './flightsDb';
