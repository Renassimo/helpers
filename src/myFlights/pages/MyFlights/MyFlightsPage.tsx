import { useEffect } from 'react';

import { NotionError } from '@/common/types/notion';
import { PageInfo, User } from '@/auth/types';
import { MyFlightData } from '@/myFlights/types';

import useAlerts from '@/common/hooks/alerts';

import PageTemplate from '@/common/templates/PageTemplate';

import FlightsProvider from '@/myFlights/providers/FlightsProvider';

import MyFlights from '@/myFlights/components/MyFlights';

const MyFlightsPage = ({
  user,
  pages,
  data,
  error,
}: {
  user: User;
  pages: PageInfo[];
  data: MyFlightData[] | null;
  error: NotionError | null;
}) => {
  const { createErrorAlert } = useAlerts();

  useEffect(() => {
    if (error) createErrorAlert(error.message || error.code || error.status);
  }, [createErrorAlert, error]);

  return (
    <FlightsProvider data={data}>
      <PageTemplate title="My Flights" user={user} pages={pages}>
        <MyFlights />
      </PageTemplate>
    </FlightsProvider>
  );
};

export default MyFlightsPage;
