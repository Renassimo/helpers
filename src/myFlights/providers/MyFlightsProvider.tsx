import { ReactNode } from 'react';

import { MyFlightData } from '@/myFlights/types';

import MyFlightsContext from '@/myFlights/contexts/MyFlightsContext';

import useMyFlightsProvider from '@/myFlights/providers/hooks/useMyFlightsProvider';

const MyFlightsProvider = ({
  children,
  data: apiData,
  nextCursor = null,
}: {
  children: ReactNode;
  data: MyFlightData[] | null;
  nextCursor: string | null;
}) => {
  const data = useMyFlightsProvider(apiData, nextCursor);

  return (
    <MyFlightsContext.Provider value={data}>
      {children}
    </MyFlightsContext.Provider>
  );
};

export default MyFlightsProvider;
