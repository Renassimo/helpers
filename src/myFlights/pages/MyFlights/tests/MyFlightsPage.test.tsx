import renderWithTheme from '@/common/tests/helpers/renderWithTheme';

import useAlerts from '@/common/hooks/alerts';

import PageTemplate from '@/common/templates/PageTemplate';
import MyFlightsProvider from '@/myFlights/providers/MyFlightsProvider';
import MyFlights from '@/myFlights/components/MyFlights';

import { mockedPageInfos, mockedUser } from '@/auth/types/mocks';
import { mockedMyFlightsList } from '@/myFlights/types/mocks';
import { mockedNotionError418 } from '@/common/types/notion/mocks';

import MockedPageTemplate from '@/common/templates/PageTemplate/mocks';
import MockedMyFlightsProvider from '@/myFlights/providers/mocks';
import MockedMyFlights from '@/myFlights/components/MyFlights/mocks';

import MyFlightsPage from '../MyFlightsPage';

jest.mock('@/common/hooks/alerts');
jest.mock('@/common/templates/PageTemplate');
jest.mock('@/myFlights/providers/MyFlightsProvider');
jest.mock('@/myFlights/components/MyFlights');

describe('MyFlightsPage', () => {
  const mockedCreateErrorAlert = jest.fn();
  const mockedUseAlerts = jest.fn(() => ({
    createErrorAlert: mockedCreateErrorAlert,
  }));

  beforeEach(() => {
    (useAlerts as unknown as jest.Mock).mockImplementation(mockedUseAlerts);
    (PageTemplate as unknown as jest.Mock).mockImplementationOnce(
      MockedPageTemplate
    );
    (MyFlightsProvider as unknown as jest.Mock).mockImplementationOnce(
      MockedMyFlightsProvider
    );
    (MyFlights as unknown as jest.Mock).mockImplementationOnce(MockedMyFlights);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders successfully', () => {
    // Arange
    // Act
    const { container } = renderWithTheme(
      <MyFlightsPage
        user={mockedUser}
        pages={mockedPageInfos}
        data={mockedMyFlightsList}
        error={null}
      />
    );
    // Assert
    expect(container).toMatchSnapshot();
    expect(mockedCreateErrorAlert).not.toHaveBeenCalled();
  });

  describe('when receives error', () => {
    test('renders successfully', () => {
      // Arange
      // Act
      const { container } = renderWithTheme(
        <MyFlightsPage
          user={mockedUser}
          pages={mockedPageInfos}
          data={null}
          error={mockedNotionError418}
        />
      );
      // Assert
      expect(container).toMatchSnapshot();
      expect(mockedCreateErrorAlert).toHaveBeenCalledWith('I am Teapot');
    });
  });
});
