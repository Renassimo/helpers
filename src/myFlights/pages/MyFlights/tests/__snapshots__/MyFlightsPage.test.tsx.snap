// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyFlightsPage renders successfully 1`] = `
<div>
  <div>
    MockedFlightsProvider - 
    <div>
      Mocked Page Template (
      My Flights
      ):
      <span>
        Jesse Pinkman
         - 
        5book
      </span>
      <div>
        MockedMyFlights
      </div>
    </div>
  </div>
</div>
`;

exports[`MyFlightsPage when receives error renders successfully 1`] = `
<div>
  <div>
    MockedFlightsProvider - 
    <div>
      Mocked Page Template (
      My Flights
      ):
      <span>
        Jesse Pinkman
         - 
        5book
      </span>
      <div>
        MockedMyFlights
      </div>
    </div>
  </div>
</div>
`;
